cardano-config/src/Cardano/Config/Orphanage.hs

```diff
@@ -60,10 +80,149 @@ instance FromJSON PortNumber where
                                  Nothing -> panic $ (show portNum)
                                                   <> " is not a valid port number."
   parseJSON invalid  = panic $ "Parsing of port number failed due to type mismatch. "
-                             <> "Encountered: " <> (T.pack $ Prelude.show invalid)
+                             <> "Encountered: " <> (Text.pack $ Prelude.show invalid)
 
 instance FromJSON Update.ApplicationName where
   parseJSON (String x) = pure $ Update.ApplicationName x
   parseJSON invalid  =
     panic $ "Parsing of application name failed due to type mismatch. "
-    <> "Encountered: " <> (T.pack $ Prelude.show invalid)
+    <> "Encountered: " <> (Text.pack $ Prelude.show invalid)
+
+instance Crypto crypto => ToJSON (ShelleyGenesis crypto) where
```

Comment:
Do we need to move this here? Could we not keep it in a separate Shelley specific module? It's quite chunky, and it is genesis specific. No objections to having "orphan" in the module name, but just because something is an orphan doesn't mean we should chuck everything in together in one big sin bin.